src/lib.rs

```diff
@@ -1,11 +1,56 @@
+extern crate unix_socket;
+
 use std::fs::File;
 use std::io;
 use std::io::Write;
+use std::io::Read;
+use std::io::Error;
+use std::u8;
+use unix_socket::UnixStream;
+
 
 // Paths to the SPI daemon sockets with incoming data from coprocessor.
 const PORT_A_UDS_PATH: &'static str = "/var/run/tessel/port_a";
 const PORT_B_UDS_PATH: &'static str = "/var/run/tessel/port_b";
 
+const MCU_MAX_SPEED: u32 = 48e6 as u32;
+// TODO: Replace with better name
+const MCU_MAX_SCL_RISE_TIME_NS: f64 = 1.5e-8 as f64;
+const MCU_MAGIC_DIV_FACTOR_FOR_I2C_BAUD: u8 = 2;
+const MCU_MAGIC_SUBTRACT_FACTOR_FOR_I2C_BAUD: u8 = 5;
+
+pub mod command {
+
+    pub const NOP: u8 = 0x00;
+    pub const FLUSH: u8 = 0x01;
+    pub const ECHO: u8 = 0x02;
+    pub const GPIO_IN: u8 = 0x03;
+    pub const GPIO_HIGH: u8 = 0x04;
+    pub const GPIO_LOW: u8 = 0x05;
+    pub const GPIO_CFG: u8 = 0x06;
+    pub const GPIO_WAIT: u8 = 0x07;
+    pub const GPIO_INT: u8 = 0x08;
+    pub const ENABLE_SPI: u8 = 0x0A;
+    pub const DISABLE_SPI: u8 = 0x0B;
+    pub const ENABLE_I2C: u8 = 0x0C;
+    pub const DISABLE_I2C: u8 = 0x0D;
+    pub const ENABLE_UART: u8 = 0x0E;
+    pub const DISABLE_UART: u8 = 0x0F;
+    pub const TX: u8 = 0x10;
+    pub const RX: u8 = 0x11;
+    pub const TXRX: u8 = 0x12;
+    pub const START: u8 = 0x13;
+    pub const STOP: u8 = 0x14;
+    pub const GPIO_TOGGLE: u8 = 0x15;
+    pub const GPIO_INPUT: u8 = 0x16;
+    pub const GPIO_RAW_READ: u8 = 0x17;
+    pub const ANALOG_READ: u8 = 0x18;
+    pub const ANALOG_WRITE: u8 = 0x19;
+    pub const GPIO_PULL: u8 = 0x1A;
+    pub const PWM_DUTY_CYCLE: u8 = 0x1B;
+    pub const PWM_PERIOD: u8 = 0x1C;
+}
```

Comment:
It could be. But it would be cast to u8 every time it's used, and isn't necessarily a complete enumeration of the possible values.